app/src/main/java/com/hedvig/app/feature/addressautocompletion/data/GetFinalDanishAddressSelectionUseCase.kt

```diff
@@ -0,0 +1,40 @@
+package com.hedvig.app.feature.addressautocompletion.data
+
+import arrow.core.Either
+import arrow.core.getOrElse
+import com.hedvig.android.owldroid.type.AddressAutocompleteType
+import com.hedvig.app.feature.addressautocompletion.model.DanishAddress
+import com.hedvig.app.util.apollo.QueryResult
+
+/**
+ * Returns a [DanishAddress] when with all inputs considered, the selection is considered a final address selection
+ * since there is no further selections that need to be made by the user.
+ */
+class GetFinalDanishAddressSelectionUseCase(
+    private val getDanishAddressAutoCompletionUseCase: GetDanishAddressAutoCompletionUseCase,
+) {
+    suspend operator fun invoke(
+        selectedAddress: DanishAddress,
+        lastSelection: DanishAddress?,
+    ): DanishAddress? {
```

Comment:
Hmm the intent of this function is to: given the current and the last address selection, figure out which address should be considered the "final" address. Whether that is because it was clicked twice or if it'd be the only possible address from all the options.
The intent of this function therefore is to have null as a normal return type, and actually it's what is returned more often than not.
I could rename the function to "getAddressOrNull" to indicate that, or make the return type a sealed class, probably will do the latter.

Reply:
Alright, I think it can be good to have separate return types for when the address is final, when its not, and when there is an error (eg. network). Then the viewmodel can decide what to do with the different results, if it want to treat error the same way as a null address for example.